docs/designers-developers/developers/tutorials/javascript/js-build-setup.md

```diff
@@ -0,0 +1,236 @@
+# JavaScript Build Setup
+
+This page covers how to set up your development environment to use the ESNext syntax. We call ESNext to JavaScript code written using features that are only available in a specification greater than ECMAScript 5 (ES5 for short) or that includes custom syntax such as [JSX](https://reactjs.org/docs/introducing-jsx.html).
+
+This documentation covers development for your plugin to work with Gutenberg, to setup a development environment to work directly on Gutenberg, see the [CONTRIBUTING.md](https://github.com/WordPress/gutenberg/blob/master/CONTRIBUTING.md) documentation.
+
+For simpler code, you can write JavaScript using ESNext and JSX syntax. This requires a transformation step to convert that syntax which a web browser may not understand, to one that it will.  Webpack and babel are the tools that perform this transformation step.
```

Comment:
The word _simpler_ at the start of this paragraph is the only hint as to _why_ a developer would adopt ESNext and would bother with a build set-up. I think we can improve the value proposition, and IMO explain upfront the trade-offs involved (i.e. here's what you get out of a build set-up, but also here's how you know it's right for you).

Reply:
I expanded on the explanation on why to use ESNext and JSX.
Take a look and let me know if there is anything additional you'd like to see.

Reply:
Thanks, looks nice.